docs/testbenches/ip_based/util_axis_fifo/index.rst

```diff
@@ -0,0 +1,237 @@
+:git-hdl:`library/util_axis_fifo <library/util_axis_fifo>` IP core.
+
+The entire HDL documentation can be found
+:external+hdl:ref:`here <util_axis_fifo>`.
+
+Block design
+-------------------------------------------------------------------------------
+
+The block design is based on the test harness with the Util AXIS FIFO IP and
+other auxiliary modules. For the ``util_axis_fifo`` IP testing, the auxiliary
+modules are a master and a slave AXI4 Stream VIP. The AXIS VIP is used to
+generate data for the DUT IP. The util_axis_fifo IP stores the data and then
+sends it to the slave AXIS VIP for verification.
+
+Block diagram
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. image:: ./util_axis_fifo.svg
+   :width: 800
+   :align: center
+   :alt: Util_AXIS_FIFO/Testbench block diagram
+
+Configuration parameters and modes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following parameters of this project that can be configured:
+
+-  ASYNC_CLK: defines the input and output clock relationship
+   Options: 0 for Synchronous or 1 for Asynchronous
+-  TKEEP_EN: enable tkeep signals
+-  TLAST_EN: enable tlast signal
+-  DATA_WIDTH: data width
+   Options: Power of 2, min 8, max 1024
+-  ADDRESS_WIDTH: address width
+   Options: 0 to 4
+-  INPUT_CLK: input clock period in ns
+   Options: 1 to 9
+-  OUTPUT_CLK: output clock period in ns
+   Options: 1 to 9
+
+Build parameters
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The parameters mentioned above can be configured when starting the build, like
+in the following example:
+
+.. shell::
+   :showuser:
+
+   $make ASYNC_CLK=1 TKEEP_EN=1 TLAST_EN=1 DATA_WIDTH=16 ADDRESS_WIDTH=3 INPUT_CLK=1 OUTPUT_CLK=3
+
+Configuration files
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The following configuration files are available:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Parameter
+     - cfg_rand
+   * - ASYNC_CLK
+     - random
+   * - TKEEP_EN
+     - random
+   * - TLAST_EN
+     - random
+   * - DATA_WIDTH
+     - random
+   * - ADDRESS_WIDTH
+     - random
+   * - INPUT_CLK
+     - random
+   * - OUTPUT_CLK
+     - random
+
+Tests
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The following test program file is available:
+
+============ ======================
+Test program Usage
+============ ======================
+test_program Tests the FIFO module.
+============ ======================
+
+Available configurations & tests combinations
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The test program is compatible with the above-mentioned configuration.
+
+CPU/Memory interconnects addresses
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Irrelevant for this testbench.
```

Comment:
We can delete the CPU/Memory interconnect subtitle if there's nothing to be mentioned. Same for Interrupts.